### PR TITLE
Remove unstable writeback FE code

### PR DESCRIPTION
**Pull request by kulyk**
* reverts #24598 as we're going to approach modal implementation a bit differently
* reverts #24051
* manually reverts #23533 as it's very prototyp-ish and breaks #24070
* removes the hack with dashboard names ending with "App". After that, we'll need to introduce Pages before we're able to put buttons onto the dashboard canvas again
* removes so-called "Basic/Standard CUD action buttons". We allowed adding a row of "New", "Edit" and "Delete" buttons to a dashboard that could be linked to a particular card. We're going to replace that with a new kind of click behavior for a single action button

### CODE REVIEW

File: frontend/src/metabase/writeback/containers/DataAppContext/DataAppContext.ts [hunk 1]
```diff
@@ -1,49 +1,24 @@
...
-type CardName = string;
-
-type ObjectDetailField = {
-  column: Column;
-  value: unknown;
-};
-
-export type FormattedObjectDetail = Record<FieldName, ObjectDetailField>;
-
-export type FormattedListInfo = { rowCount: { value?: number } };
```
Comment (iethree) [inline frontend/src/metabase/writeback/containers/DataAppContext/DataAppContext.ts:16]: these need to be removed from the import in `utils.js`

**Review by iethree — APPROVED** [review]
Flagged one little import issue. Weird, that my local `yarn type-check` doesn't catch it but CI did 🤔

File: frontend/src/metabase/writeback/containers/DataAppContext/DataAppContext.ts (hunk 1 above)
Comment (kulyk) [reply frontend/src/metabase/writeback/containers/DataAppContext/DataAppContext.ts:16]: Yup, that's in progress, thanks for pointing out

Comment (iethree) [reply frontend/src/metabase/writeback/containers/DataAppContext/DataAppContext.ts:16]: oops. didn't see this was still draft. 😊

**Review by iethree — APPROVED** [review]
🚀

**Pull request merged**